component/component/main.jsonnet

```diff
@@ -0,0 +1,59 @@
+// main template for openshift4-service-mesh
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.openshift4_service_mesh;
+
+local needs_es_operator =
+  local logging_params = std.get(inv.parameters, 'openshift4_logging', {
+    components: {
+      elasticsearch: {
+        // if `openshift4_logging` key not in parameters, we always need to
+        // install es operator.
+        enabled: false,
+      },
+    },
+  });
+  !logging_params.components.elasticsearch.enabled;
+
+local operatorlib = import 'lib/openshift4-operators.libsonnet';
+
+// Define outputs below
+{
+  namespace: kube.Namespace(params.namespace) {
+    metadata+: {
+      annotations+: {
+        // allow istio cni pods to be scheduled on all nodes.
+        // TODO(sg): might be nicer to have a tighter node selector, but
+        // removing the annotation completely to restrict the operator to app
+        // nodes changes the daemonset behavior and messes up pod scheduling.
+        'openshift.io/node-selector': '',
+      },
+      labels+: {
+        // include namespace in cluster monitoring
+        'openshift.io/cluster-monitoring': 'true',
+        // ignore namespace in user-workload monitoring
+        'openshift.io/user-monitoring': 'false',
+      },
+    },
+  },
+  [if needs_es_operator then 'es_operator']: [
+    operatorlib.managedSubscription(
+      'openshift-operators-redhat',
```

Comment:
The official docs for installing EFK logging () say:

> You must specify the openshift-operators-redhat namespace. The openshift-operators namespace might contain Community Operators, which are untrusted and could publish a metric with the same name as an OpenShift Container Platform metric, which would cause conflicts.

I'll see if the operator works correctly when it's installed in its own namespace

Reply:
seems to work fine, so the ES operator is now installed in namespace `<namespace>-es-operator`, where `<namespace>` is the namespace configured for the component.